sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -234,8 +249,8 @@ public BinaryData getBody() {
                 return BinaryData.fromBytes(amqpAnnotatedMessage.getBody().getFirstData());
             case SEQUENCE:
             case VALUE:
-                throw logger.logExceptionAsError(new UnsupportedOperationException("Not supported AmqpBodyType: "
-                    + type.toString()));
+                throw logger.logExceptionAsError(new IllegalStateException("Message  body type is not DATA, instead "
+                    + "it is: " + type.toString()));
```

Comment:
Looks like this is still not fixed.  Why do we throw this exception when SEQUENCE and VALUE are now supported formats.

Reply:
This method `getBody()` represents AMQP Type `DATA` .  If the message body type is `SEQUENCE` or `VALUE`, user should call `getSequence()` and `getValue()`on `AmqpMessgeBody`. 
So here we still need to throw this error.